src/benchmark/RemotePingPong.NET46/Program.cs

```diff
@@ -0,0 +1,282 @@
+                {
+                    var result1 = await Benchmark(throughput, repeat, bestThroughput, redCount);
+                    bestThroughput = result1.Item2;
+                    redCount = result1.Item3;
+                }
+            }
+
+            Console.ForegroundColor = ConsoleColor.Gray;
+            Console.WriteLine("Done..");
+        }
+
+        public static IEnumerable<int> GetClientSettings()
+        {
+            yield return 1;
+            yield return 5;
+            yield return 10;
+            yield return 15;
+            yield return 20;
+            yield return 25;
+            yield return 30;
+        }
+
+        private static long GetTotalMessagesReceived(int numberOfClients, long numberOfRepeats)
+        {
+            return numberOfClients * numberOfRepeats * 2;
+        }
+
+        private static async Task<Tuple<bool, long, int>> Benchmark(int numberOfClients, long numberOfRepeats, long bestThroughput, int redCount)
+        {
+            var totalMessagesReceived = GetTotalMessagesReceived(numberOfClients, numberOfRepeats);
+            var system1 = ActorSystem.Create("SystemA", CreateActorSystemConfig("SystemA", "127.0.0.1", 0));
+
+            var system2 = ActorSystem.Create("SystemB", CreateActorSystemConfig("SystemB", "127.0.0.1", 0));
+
+            List<Task<long>> tasks = new List<Task<long>>();
+            List<IActorRef> receivers = new List<IActorRef>();
+
+            var canStart = system1.ActorOf(Props.Create(() => new AllStartedActor()), "canStart");
+
+            var system1Address = ((ExtendedActorSystem)system1).Provider.DefaultAddress;
+            var system2Address = ((ExtendedActorSystem)system2).Provider.DefaultAddress;
+
+            var echoProps = Props.Create(() => new EchoActor()).WithDeploy(new Deploy(new RemoteScope(system2Address)));
+
+            for (var i = 0; i < numberOfClients; i++)
+            {
+                var echo = system1.ActorOf(echoProps, "echo" + i);
+                var ts = new TaskCompletionSource<long>();
+                tasks.Add(ts.Task);
+                var receiver =
+                    system1.ActorOf(
+                        Props.Create(() => new BenchmarkActor(numberOfRepeats, ts, echo)),
+                        "benchmark" + i);
+
+                receivers.Add(receiver);
+
+                canStart.Tell(echo);
+                canStart.Tell(receiver);
+            }
+
+            var rsp = await canStart.Ask(new AllStartedActor.AllStarted(), TimeSpan.FromSeconds(10));
+            var testReady = (bool)rsp;
+            if (!testReady)
+            {
+                throw new Exception("Received report that 1 or more remote actor is unable to begin the test. Aborting run.");
+            }
+
+            var sw = Stopwatch.StartNew();
+            receivers.ForEach(c =>
+            {
+                for (var i = 0; i < 50; i++) // prime the pump so EndpointWriters can take advantage of their batching model
```

Comment:
Kick off a batch of messages locally at the start of each test - allows the `EndpointWriter` batching system to run the same way it would in a production environment.